File: python_implementation/john_linear_strategy.py
# John's strategy = sequential search
def find_number_sequentially():  # define the function find_number_sequentially
    """
    Finds a target number in a list using sequential search.
    """
    # variables
    start_range = int(input("Enter the starting range: "))
    end_range = int(input("Enter the ending range: "))  # variables
    target_number = int(input("Enter the target number: "))

    number_list = list(range(start_range, end_range + 1))

    if target_number not in number_list:
        print("Target number not found in the specified range.")
        # print("The target number is in the specified range of value.")
        return

    tries = 0
    for number in number_list:
        tries += 1  # tries = tries + 1
        if number == target_number:
            print(f"Target number {target_number} found in {tries} tries.")
            return


# Example usage
find_number_sequentially()

File: c_implementation/john_linear_strategy.c
#include <stdio.h>

// Function to find a target number using sequential search
void find_number_sequentially() {
    int start_range, end_range, target_number;

    // Input the range and target number
    printf("Enter the starting range: ");
    scanf("%d", &start_range);
    printf("Enter the ending range: ");
    scanf("%d", &end_range);
    printf("Enter the target number: ");
    scanf("%d", &target_number);

    int tries = 0;
    for (int i = start_range; i <= end_range; i++) {
        tries++;
        if (i == target_number) {
            printf("Target number %d found in %d tries.\n", target_number, tries);
            return;
        }
    }

    // If the loop completes, the target was not found
    printf("Target number not found in the specified range.\n");
}

int main() {
    find_number_sequentially();
    return 0;
}
